api/envoy/api/v2/core/base.proto

```diff
@@ -76,10 +77,43 @@ message Locality {
   string sub_zone = 3;
 }
 
+// BuildVersion combines SemVer version of extension with free-form build information
+// (i.e. 'alpha', 'private-build') as a set of strings.
+message BuildVersion {
+  // SemVer version of extension
+  type.SemanticVersion version = 1;
+
+  // Free-form build information i.e. 'alpha'
+  repeated string labels = 2;
+}
+
+// Version and identification for an Envoy extension.
+message Extension {
+  // This is the name of the Envoy filter as specified in the Envoy
+  // configuration, e.g. envoy.router, com.acme.widget.
+  string name = 1;
+
+  // Category of the extension
+  string category = 2;
```

Comment:
Just to clarify, I would assume specific type/name of the WASM filter would go into the category as well? Because its name would be a generic "envoy.wasm" in all cases.

Reply:
This field is populated from the factory's category() method. I've read through the #9358 and I'm still clear on what needs to happen for wasm filters. Perhaps WASM filter category should be supplied with the config somehow?

Reply:
@PiotrSikora could you please chime in on how WASM could be reporting its specific kind through this API?

Reply:
Also @jplevyak. I think that we can add some higher-order category support later for things like WASM. Ultimately I'd like to see how this plays out, but we shouldn't block.

Reply:
Wasm doesn't use protobuf for configuration. Do we need an alternative way to categorize / type the Wasm filters?

Reply:
Hmm, this would be rather unfortunate to add separate categorization for wasm extensions. What  does wasm extensions use for configuration? Can they provide an identity config proto?

Reply:
Wasm just uses an opaque blob. The reason being protobuf limits the choice of languages (and it's pretty slow in general).

Reply:
I think we can support Wasm additively after this PR ships. While the Wasm extension providing the runtime is a true Envoy extension, its plugins will be their own thing and probably need a structured reporting mechanism. I don't think we break that by going with what is in this PR.

Reply:
+1 I think wasm can fit into this scheme by having all its plugin to have the same category i.e. "wasm" and add thread safe implementation of the FactoryRegistryProxy that can provide names of available wasm plugins. The name can come from config metadata.

Reply:
This should be fine for now, but I hope wasm will develop some first-class integration with xDS/Envoy version reporting mechanism, rather just leaving it for generic "wasm" and then providing opaque blob for the rest. That's a note for wasm in general, not specifically for this implementation.